opencog/query/PatternTerm.h

```diff
@@ -54,14 +54,16 @@ namespace opencog {
 
 class PatternTerm;
 typedef std::shared_ptr<PatternTerm> PatternTermPtr;
+typedef std::weak_ptr<PatternTerm> PatternTermWPtr;
 typedef std::vector<PatternTermPtr> PatternTermSeq;
+typedef std::vector<PatternTermWPtr> PatternTermWSeq;
 
 class PatternTerm
 {
 	protected:
 		Handle _handle;
 		PatternTermPtr _parent;
-		PatternTermSeq _outgoing;
+		PatternTermWSeq _outgoing;
```

Comment:
It probably would have been easier to make the parent be weak.  Or, at least, that would make it consistent with the atomspace, where down-pointers are strong, and the up-ones (incoming set) are weak.